ops/model.py

```diff
@@ -663,3 +664,40 @@ def network_get(self, endpoint_name, relation_id=None):
             if 'relation not found' in str(e):
                 raise RelationNotFoundError() from e
             raise
+
+    def add_metrics(self, metrics, labels=None):
+        cmd = ['add-metric']
+        key_re = re.compile(r'^[a-zA-Z](?:[a-zA-Z0-9-_]*[a-zA-Z0-9])?$')
+
+        def is_valid_metric_value(value):
+            try:
+                float(value)
+                return True
+            except ValueError:
+                return False
+
+        def is_valid_label_value(value):
+            # Label values cannot be empty, contain commas or equal signs as those are used by add-metric as separators.
+            return value and all(c not in str(value) for c in ',=')
+
+        if labels:
+            label_args = []
+            for k, v in labels.items():
+                if not key_re.match(k):
+                    raise ModelError(f'invalid label key "{repr(k)}": must start with [a-zA-Z] and contain [a-zA-Z0-9_] only')
```

Comment:
The repr will add its own quotes so you need to drop the outer ones,
```suggestion
                    raise ModelError(f'invalid label key {repr(k)}: must start with [a-zA-Z] and contain [a-zA-Z0-9_] only')
```